File: 2_Btree/JavaScript/main.js
var antlr4 = require('antlr4');
var BtreeLexer = require('./BtreeLexer').BtreeLexer;
var BtreeParser = require('./BtreeParser').BtreeParser;

var input = "(0(1()())(2()()))"
var chars = new antlr4.InputStream(input);

// Lexer step
var lexer = new BtreeLexer(chars);
var tokens  = new antlr4.CommonTokenStream(lexer);

// Print the tokens readed
tokens.fill();
tokens.getTokens(0, input.length).forEach(token =>
    console.log(token.toString())
)

// Parser step
var parser = new BtreeParser(tokens);
parser.buildParseTrees = true;
var tree = parser.parse_all();

// Print the parsing tree
if(tree.parser._syntaxErrors == 0){
    console.log("Parsing result = SUCCESS")
    console.log(tree.toStringTree(null, parser))
}

File: 2_Btree/Java/Main.java
import org.antlr.v4.runtime.*;
import org.antlr.v4.runtime.tree.*;
import java.io.FileInputStream;
import java.io.IOException;

public class Main {
  public static void main(String args[]) throws IOException{
    // Reads from file
    FileInputStream input_stream = new FileInputStream(args[0]);
    CharStream input = CharStreams.fromStream(input_stream); 
    
    // Creates the Lexer
    BtreeLexer lexer = new BtreeLexer(input);
    CommonTokenStream tokens = new CommonTokenStream(lexer); 

    // Print the tokens readed
    tokens.fill();
    for(Token token : tokens.getTokens()){
        System.out.println(token.toString());
    }
    System.out.println();

    // Create the parser
    BtreeParser parser = new BtreeParser(tokens);
    ParseTree tree = parser.parse_all();

    if(parser.getNumberOfSyntaxErrors() == 0){
      System.out.println("Parsing result = SUCCESS\n");
      // Print the parsing tree
      System.out.println(tree.toStringTree(parser));
      System.out.println();
    }
  }
}